opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/TestUtils.java

```diff
@@ -13,26 +13,44 @@
  */
 package io.opentracing.contrib.spring.cloud;
 
-import java.util.Collection;
-
+import io.opentracing.Tracer;
 import io.opentracing.mock.MockSpan;
+import io.opentracing.util.GlobalTracer;
 import org.junit.Assert;
 
+import java.lang.reflect.Field;
+import java.util.Collection;
+
 /**
  * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
  */
 public class TestUtils {
-  /**
-   * Check if all spans have same traceId.
-   *
-   * @param spans the spans to check
-   */
-  public static void assertSameTraceId(Collection<MockSpan> spans) {
-    if (!spans.isEmpty()) {
-      final long traceId = spans.iterator().next().context().traceId();
-      for (MockSpan span : spans) {
-        Assert.assertEquals(traceId, span.context().traceId());
-      }
+    /**
+     * Check if all spans have same traceId.
+     *
+     * @param spans the spans to check
+     */
+    public static void assertSameTraceId(Collection<MockSpan> spans) {
+        if (!spans.isEmpty()) {
+            final long traceId = spans.iterator().next().context().traceId();
+            for (MockSpan span : spans) {
+                Assert.assertEquals(traceId, span.context().traceId());
+            }
+        }
+    }
+
+    /**
+     * Set tracer in GlobalTracer
+     * @param tracer tracer
+     */
+    public static void setGlobal(Tracer tracer) {
```

Comment:
Can we make this private and move it to `MockTracingConfiguration` and set the tracer right after the `MockTracer` is created?

It will ensure that global tracer equals to tracer bean in the context.

Reply:
somehow it doesn't work. test started to fail

Reply:
Does the test fail or does it fail on different runtime exception?  Try to only set the tracer to noop e.g. set it to the initial state.

Reply:
Feign tests fails:
```
org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in io.opentracing.contrib.spring.cloud.feign.FeignTest that uses io.opentracing.mock.MockTracer was not fulfilled within 10 seconds.

```

Reply:
fixed